spiketoolkit/comparison/groundtruthcomparison.py

```diff
@@ -0,0 +1,392 @@
+                                    delta_frames=delta_frames, min_accuracy=min_accuracy, n_jobs=n_jobs, verbose=verbose)
+        self.exhaustive_gt = exhaustive_gt
+
+        self._do_count()
+
+    
+    def _do_count(self):
+        """
+        Do raw count into a dataframe.
+        """
+        unit1_ids = self._sorting1.get_unit_ids()
+        columns = ['tp', 'fn', 'cl','fp', 'num_gt', 'num_other', 'other_id']
+        self.count = pd.DataFrame(index=unit1_ids, columns=columns)
+        for u1 in unit1_ids:
+            u2 = self._unit_map12[u1]
+            
+            self.count.loc[u1, 'tp'] = np.sum(self._labels_st1[u1] == 'TP')
+            self.count.loc[u1, 'cl'] = sum(e.startswith('CL') for e in self._labels_st1[u1])
+            self.count.loc[u1, 'fn'] = np.sum(self._labels_st1[u1] == 'FN')
+            self.count.loc[u1, 'num_gt'] = self._labels_st1[u1].size
+            self.count.loc[u1, 'other_id'] = u2
+
+            if u2==-1:
+                self.count.loc[u1, 'fp'] = 0
+                self.count.loc[u1, 'num_other'] = 0
+            else:
+                self.count.loc[u1, 'fp'] = np.sum(self._labels_st2[u2] == 'FP')
+                self.count.loc[u1, 'num_other'] = self._labels_st2[u2].size
+            
+
+    def get_performance(self, method='by_spiketrain', output='pandas'):
+        """
+        Get performance rate with several method:
+          * 'raw_count' : just render the raw count table
+          * 'by_spiketrain' : render perf as rate spiketrain by spiketrain of the GT
+          * 'pooled_with_sum' : pool all spike with a sum and compute rate
+          * 'pooled_with_average' : compute rate spiketrain by spiketrain and average
+
+        Parameters
+        ----------
+        method: str
+            'by_spiketrain', 'pooled_with_sum' or 'pooled_with_average'
+        output: str
+            'pandas' or 'dict'
+
+        Returns
+        -------
+        perf: pandas dataframe/series (or dict)
+            dataframe/series (based on 'output') with performance entries
+        """
+        possibles = ('raw_count', 'by_spiketrain', 'pooled_with_sum', 'pooled_with_average')
+        if method not in possibles:
+            raise Exception("'method' can be " + ' or '.join(possibles))
+
+
+        if method =='raw_count':
+            perf = self.count
+            
+        elif method == 'by_spiketrain':
+            unit1_ids = self._sorting1.get_unit_ids()
+            perf = pd.DataFrame(index=unit1_ids, columns=_perf_keys)
+            
+            counts = self.count
+            perf['tp_rate'] = counts['tp'] / counts['num_gt']
+            perf['cl_rate'] = counts['cl'] / counts['num_gt']
+            perf['fn_rate'] = counts['fn'] / counts['num_gt']
+            perf['fp_rate'] = counts['fp'] / counts['num_gt']
+            
+            perf['accuracy'] = perf['tp_rate'] / (perf['tp_rate'] + perf['fn_rate']+perf['fp_rate'])
```

Comment:
in our case `accuracy` and `tp_rate` is the same, no?

num_gt = num_tp + num_fn + num_fp --> 
accuracy = (num_tp / num_gt) / ((num_tp / num_gt) + (num_fn / num_gt) + (num_fp / num_gt)) = 
= num_tp / (num_tp + num_fp + num_fn) = num_tp / num_gt = tp_rate

@samuelgarcia @colehurwitz31 @mhhennig  right?

Reply:
Yes you are right.
Do we have the same definition as Jeremy website ?